### ISSUE TITLE
Entire buffer cleared when file changes causing scroll position to be lost

### ISSUE THREAD

**Issue by adamreisnz** [atom/atom#13656]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
When using linters with auto-fix enabled (for example linter-eslint), it will fix code style errors for you upon saving the file you're working on, e.g. missing semi colon, incorrect indentation, etc.

For a while now, when this happens, Atom tends to momentarily blank out the window and then re-insert the fixed contents, causing the window to scroll up. This is problematic, as it causes you to lose your scroll position and as such your focus of the area of code you were working on.

### Steps to Reproduce

1. Install/enable only core packages and linter as well as linter-eslint
2. Use the following rules for eslint:

```
{
    "parser": "babel-eslint",
    "env": {
        "es6": true
    },
    "rules": {
        "semi": [ 1, "always" ]
    }
}
```

3. Open these file contents (thanks @dwelle) http://pastebin.com/raw/Yk6K1w5k and save as `test.js`.
4. Scroll to the bottom and remove the last semi colon.
5. Save the file

**Expected behavior:**
Contents remains visible, then is fixed/replaced by a fixed contents, while maintaining scroll position.

**Actual behavior:**
The buffer is cleared, then contents re-appears and the scroll position is at the top.

**Reproduces how often:**
Nearly 100% with the provided example file.

### Versions
OS X: 10.11.5
Atom    : 1.14.0-dev-157e8e1
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
apm  1.15.1
npm  3.10.5
node 4.4.5 x64
python 2.7.10
git 2.6.2

### Additional information
Note that while the example file is quite large with 3000+ lines, this behaviour *does* occur quite regularly with smaller files as well. It's just not 100% reproducible in those cases. But it could be that the file size/length emphasises the issue.

### See also
- https://github.com/AtomLinter/linter-eslint/issues/620
- https://github.com/AtomLinter/linter-eslint/issues/774
- https://github.com/atom/atom/issues/11483

**Comment by helsont** [comment]
I see this issue occur all the time with files less than 400 lines long. With respect to reproducibility, I'm not sure, but in terms of frequency of the issue, it's quite frequent.

**Comment by mmcken3** [comment]
I am running Atom 1.13.1 and Mac OS X 10.10.5. Using the exact steps above I was able to reproduce this issue 9/10 times.

**Comment by adamreisnz** [comment]
I think this has been fixed as of 1.19, closing.
